internal/rbd/nodeserver.go

```diff
@@ -187,33 +185,13 @@ func populateRbdVol(
 		}
 		rv.RbdImageName = volID
 	} else {
-		var vi util.CSIIdentifier
-		var imageAttributes *journal.ImageAttributes
-		err = vi.DecomposeCSIID(volID)
+		rv, err = GenVolFromVolID(ctx, volID, cr, req.GetSecrets())
 		if err != nil {
-			err = fmt.Errorf("error decoding volume ID (%s): %w", volID, err)
-
-			return nil, status.Error(codes.Internal, err.Error())
-		}
-
-		j, err = volJournal.Connect(rv.Monitors, rv.RadosNamespace, cr)
-		if err != nil {
-			log.ErrorLog(ctx, "failed to establish cluster connection: %v", err)
-
-			return nil, status.Error(codes.Internal, err.Error())
-		}
-		defer j.Destroy()
-
-		imageAttributes, err = j.GetImageAttributes(
-			ctx, rv.Pool, vi.ObjectUUID, false)
-		if err != nil {
-			err = fmt.Errorf("error fetching image attributes for volume ID (%s): %w", volID, err)
+			rv.Destroy()
+			err = fmt.Errorf("error generating volume (%s): %w", volID, err)
```

Comment:
why assign `err` and not use `status.Errorf()` below?

Reply:
Yes, that can be done, Updated to use status.Errorf